### PR TITLE
Interact Menu - Combine parent menu with single child

### PR DESCRIPTION
**Pull request by Dystopian**
**When merged this pull request will:**
- add setting to combine parent menu with single child;
- adapt `fastroping` `Cut ropes` action for the setting;
- clean up `chemlights` and `map` menu configs a little.

Setting is disabled by default. ~Still not sure about setting name and description.~

### CODE REVIEW

**Review by veteran29 — APPROVED** [review]
works. Nice feature 👍

**Comment by PabstMirror** [comment]
Really minor, but I want to test how this will work with https://github.com/acemod/ACE3/pull/6876
where I used the child as a "safety"
I think we can just add a dummy `statement` if it's a problem

File: addons/interact_menu/stringtable.xml [hunk 1]
```diff
@@ -489,5 +489,9 @@
             <Spanish>Selector de color</Spanish>
             <Turkish>Seçici Renk</Turkish>
         </Key>
+        <Key ID="STR_ACE_Interact_Menu_combineParentWithSingleChild">
+            <English>Combine parent menu with single child item</English>
```
Comment (mharis001) [inline addons/interact_menu/stringtable.xml:493]: I think title case for setting display names, maybe something like: "Consolidate Single Child Actions". And then the description can be: "Combines parent actions with only one child action together."

Comment (Dystopian) [reply addons/interact_menu/stringtable.xml]: > "Combines parent actions with only one child action together."

I don't think camel case is good for settings especially for those looking like sentences. Also I replaced `parent actions` with `parent action`.

File: addons/interact_menu/functions/fnc_collectActiveActionTree.sqf [hunk 2]
```diff
@@ -91,5 +101,22 @@ if ((_activeChildren isEqualTo []) && {_statementCode isEqualTo {}}) exitWith {
     []
 };
 
+if (GVAR(consolidateSingleChild) && {count _activeChildren == 1} && {_statementCode isEqualTo {}}) then {
+    _activeChildren select 0 params ["_childActionData", "_childChildren"];
```
Comment (PabstMirror) [inline addons/interact_menu/functions/fnc_collectActiveActionTree.sqf:105]:
```suggestion
    _activeChildren select 0 params ["_childActionData", "_childChildren", "_childObject"];
```

File: addons/interact_menu/functions/fnc_collectActiveActionTree.sqf [hunk 3]
```diff
@@ -91,5 +101,22 @@ if ((_activeChildren isEqualTo []) && {_statementCode isEqualTo {}}) exitWith {
...
+        _iconChild,
+        _statementChild,
+        _conditionCode,
+        _insertChildrenCode,
+        _customParamsChild,
+        _position,
+        _distance,
+        _paramsChild
+    ];
+    _activeChildren = _childChildren;
```
Comment (PabstMirror) [inline addons/interact_menu/functions/fnc_collectActiveActionTree.sqf:119]:
```suggestion
    _activeChildren = _childChildren;
    _object = _childObject;
```

**Review by PabstMirror — CHANGES_REQUESTED** [review]
We need this if child action has different object
e.g. a single passenger won't show any interactions because the object was still on the vehicle

**Review by PabstMirror — APPROVED** [review]

**Comment by Dystopian** [comment]
@PabstMirror please review the last commit

**Review by PabstMirror — APPROVED** [review]

**Pull request merged**